.github/workflows/build-push-dockerhub.yaml

```diff
@@ -0,0 +1,24 @@
+---
+name: build-and-publish-dockerhub
+on:
+  push:
+    branches:
+      - main
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: build image
+        shell: bash
+        run: |
+          docker build -t jackdwyer/bom_exporter:latest .
+      - name: login to dockerhub
+        shell: bash
+        run: |
+          docker login --username ${{ secrets.DOCKERHUB_USERNAME }} --password ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: publish to dockerhub
+        shell: bash
+        run: |
+          docker push jackdwyer/bom_exporter:latest
```

Comment:
Here too, DOCKERHUB_TAG.
And also, perhaps consider `docker/build-push-action`?
https://github.com/marketplace/actions/build-and-push-docker-images